File: Data Structures & Algorithms/find-duplicate-integer/submission-3.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        for i in range(1, len(nums)):
            if nums[abs(nums[i]) - 1] < 0:
                return abs(nums[i])
            nums[abs(nums[i]) - 1] *= -1
        print(nums)
        return abs(nums[0])

File: Data Structures & Algorithms/min-cost-to-connect-points/submission-3.cpp
class Solution {
public:
    int minCostConnectPoints(vector<vector<int>>& points) {
       int n = points.size(),  node = 0;
       vector<int> dist(n, INT_MAX);
       vector<bool> visit(n, false);
        int edges = 0, res = 0;

        while (edges < n - 1) {
            visit[node] = true;
            int nextNode = -1;
            for (int i = 0; i < n; ++i) {
                if (visit[i]) continue;
                int currDist = abs(points[i][0] - points[node][0])
                     + abs(points[i][1] - points[node][1]);
                dist[i] = min(dist[i], currDist);
                if (nextNode == -1 || dist[i] < dist[nextNode])
                    nextNode = i;
            }
            res += dist[nextNode];
            node = nextNode;
            edges++;
        }
        return res;
    }
};
